subsys/bluetooth/audio/has.c

```diff
@@ -1200,6 +1289,84 @@ int bt_has_register(const struct bt_has_register_param *param)
 		has.features |= BT_HAS_FEAT_WRITABLE_PRESETS_SUPP;
 	}
 
+	return 0;
+}
+
+int bt_has_features_set(const struct bt_has_features_param *features)
+{
+	uint8_t tmp_features;
+	int err;
+
+	/* Check whether any features will change, otherwise we don't want to notify clients */
+	if (!((has.features & BT_HAS_FEAT_HEARING_AID_TYPE_LSB) ^
+	      (features->type & BT_HAS_FEAT_HEARING_AID_TYPE_LSB) ||
+	      (has.features & BT_HAS_FEAT_HEARING_AID_TYPE_MSB) ^
+		  (features->type & BT_HAS_FEAT_HEARING_AID_TYPE_MSB) ||
+	      (has.features & BT_HAS_FEAT_PRESET_SYNC_SUPP) ^
+		  features->preset_sync_support ||
+	      (has.features & BT_HAS_FEAT_INDEPENDENT_PRESETS) ^
+		  features->independent_presets)) {
```

Comment:
Holy batman this is difficult to follow :D 

I strongly suggest to move these into variables, functions or macro to improve readability

Reply:
Haha, fair enough